lib/resolve-decl.js

```diff
@@ -9,16 +12,91 @@ var shallowCloneNode = require('./shallow-clone-node');
 var findNodeAncestorWithSelector = require('./find-node-ancestor-with-selector');
 var cloneSpliceParentOntoNodeWhen = require('./clone-splice-parent-onto-node-when');
 
+var parser = require('postcss-selector-parser');
+
+function equals(a, b) {
+	return a && b && a.type !== undefined && a.value !== undefined && a.type === b.type && a.value === b.value;
+}
+
+function compatible(a, b) {
+	var af = a.filter(c => !b.find(d => equals(d, c)));
+	var bf = b.filter(c => !a.find(d => equals(d, c)));
+	var ac = a.find(c => c.type == 'combinator');
+	var bc = b.find(c => c.type == 'combinator');
+
+	if(bf.length == 0 && ((!ac && !bc) || equals(ac, bc)))
+	{
+		return true;
+	}
+
+	return false;
+}
+
+function compatible2(a, b) {
+	var af = a.filter(c => !b.find(d => equals(d, c)));
+	var bf = b.filter(c => !a.find(d => equals(d, c)));
 
+	if(bf.length == 0 && af.length != 0)
+	{
+		return true;
+	}
+
+	return false;
+}
```

Comment:
Let's add a comment describing how `compatible` and `compatible2` are different and when you should use either